fili-core/src/test/groovy/com/yahoo/bard/webservice/data/SerializationResources.groovy

```diff
@@ -144,35 +143,36 @@ class SerializationResources extends Specification {
         schema3 = buildSchema(baseSchemaTypeMap)
 
         List<Result> results = new ArrayList<>([result1, result2])
-        resultSet = new ResultSet(results, schema)
+        resultSet = new ResultSet(schema, results)
 
-        DateTime ny = new DateTime(2011, 2, 2, 7, 0, 0, 0, DateTimeZone.forID("UTC"));
-        DateTime la = new DateTime(2011, 2, 3, 10, 15, 0, 0, DateTimeZone.forID("UTC"));
+        DateTime ny = new DateTime(2011, 2, 2, 7, 0, 0, 0, DateTimeZone.forID("UTC"))
+        DateTime la = new DateTime(2011, 2, 3, 10, 15, 0, 0, DateTimeZone.forID("UTC"))
         interval = new Interval(ny, la)
         bigDecimal = new BigDecimal("100")
 
         responseContext = new ResponseContext([:])
         responseContext.put("randomHeader", "someHeader")
-        responseContext.put("missingIntervals", ["a","b","c", new SimplifiedIntervalList([interval]), bigDecimal])
+        responseContext.put("missingIntervals", ["a","b","c", new SimplifiedIntervalList([interval]), bigDecimal] as ArrayList)
 
         responseContext1 = new ResponseContext([:])
         responseContext1.put("randomHeader", "someHeader")
-        responseContext1.put("apiMetricColumnNames", ["metric1, metric2"] as Set)
+        responseContext1.put("apiMetricColumnNames", ["metric1, metric2"] as LinkedHashSet)
```

Comment:
Can this `as LinkedHashSet` solve the the problem where we have to use `Iterator` instead of `Set`?